crates/ra_editor/src/extend_selection.rs

```diff
@@ -36,6 +42,24 @@ pub(crate) fn extend(root: SyntaxNodeRef, range: TextRange) -> Option<TextRange>
     }
 }
 
+fn extend_single_word_in_comment(leaf: SyntaxNodeRef, range: TextRange) -> Option<TextRange> {
+    let text : &str = leaf.leaf_text().unwrap();
+    let cursor_position: u32 = (range.start() - leaf.range().start()).into();
+
+    let (before, after) = text.split_at(cursor_position as usize);
+    let start_idx = before.rfind(char::is_whitespace);
+    let end_idx = after.find(char::is_whitespace);
+
+    match (start_idx, end_idx) {
+        (Some(start), Some(end)) => {
+            let from : TextUnit = (start as u32 + 1).into();
+            let to : TextUnit = (cursor_position + (end as u32)).into();
+            Some(TextRange::from_to(from, to))
+        },
+        (_, _) => None
+    }
+}
```

Comment:
Hm, this switching between `TextUnit` and `usize` via `u32` does not look pretty at all. I think I'll need to extend `TextUnit` API to make it less horrible.

The reason why we use `TextUnit` and `TextRange` instead of `usize` and `(usize, usize)` is that, for higher-level API, it's useful to distinguish text offsets from some random indices, and we also want to use a compact `u32` internal representationt.

Reply:
That makes sense. I pulled the casts out to the initialization of `start_idx` and `end_idx`.